website/static/js/contribAdder.js

```diff
@@ -313,6 +344,33 @@ var AddContributorViewModel = oop.extend(Paginator, {
         }
         return false;
     },
+    selectAllNodes: function() {
+        //select all nodes to add a contributor to.  THe changed variable is set here for timing between
+        // treebeard and knockout
+        var self = this;
+        var nodesState = ko.toJS(self.nodesState());
+        for (var key in nodesState) {
+            if (nodesState[key].enabled) {
+                nodesState[key].checked = true;
+                nodesState[key].changed = nodesState[key].checked !== self.nodesOriginal[key].checked;
+            }
+        }
+        self.nodesState(nodesState);
+    },
+    selectNoNodes: function() {
+        //select no nodes to add a contributor to.  THe changed variable is set here for timing between
+        // treebeard and knockout
+        var self = this;
+        var nodesState = ko.toJS(self.nodesState());
+        for (var key in nodesState) {
+            if (nodesState[key].enabled && nodesState[key].checked) {
+                nodesState[key].checked = false;
+                nodesState[key].changed = nodesState[key].checked !== self.nodesOriginal[key].checked;
+            }
+        }
+        self.nodesState(nodesState);
+        m.redraw(true);
```

Comment:
Is this redraw necessary, given that you have a subscriber function on `nodesState` that calls `m.redraw`?